### ISSUE TITLE
[BETA16] Ticking Entity client crashes

### ISSUE THREAD

**Issue by Iastre** [Azanor/thaumcraft-beta#1004]
Certain scenarios cause constant crashes due to ticking entities. Particularly bad are various mobs from alexthe666's Ice & Fire mod, which works fine when Thaumcraft is not present. These crashes occur in a client connected to a remote server, they do not appear to occur at all in SP.

After extensive troubleshooting and process of elimination, it appears that these crashes only occur when Thaumcraft is installed, and only started since the update to BETA16. I can't say for sure that the bug is in Thaumcraft, but it does seem to be implicated.

The crash seems to be related to trying to cast an Integer as a Boolean.

[crash-2018-06-10_20.47.50-client.txt](https://github.com/Azanor/thaumcraft-beta/files/2088407/crash-2018-06-10_20.47.50-client.txt)

**Comment by Iastre** [comment]
This issue has been reported in several bug reports on the ice and fire repo and they will hopefully address the problem as well. I chose to mention it here because it hasn't been brought up at all and the crashes started when I updated to BETA16 (the only real change I am aware of in my build) and they persist as long as I have BETA16 installed, disappearing when I roll back to an earlier version of Thaumcraft or remove it entirely.

I also don't know if the only issues occur between Ice & Fire and Thaumcraft. One or the other of these mods may have the same issue with other mods as well. The fact that it does not appear in the stack trace means that most people experiencing this problem wouldn't be able to trace it back to Thaumcraft, even if Thaumcraft is indeed at fault.
